hep/hep0.md

```diff
@@ -0,0 +1,105 @@
+
+#### Review and Resolution
+A HoloViz core developer will then formally review the pull request for format, scope, and evidence of consensus. They may also notify other interested parties to weigh in on the HEP. If consensus has been reached, the HoloViz core developer will update the status of the pull request, assign a HEP number, add a link to the discussion in the [HEP table](accepted_heps.md), and merge it. All HEPs will be resolved as either 'rejected', 'accepted', or 'deferred' depending on the consensus of the community.
+
+In unusual and controversial cases where consensus has not been possible to achieve, a HoloViz core developer may ask the steering committee to take a HEP up for vote, following the policies described in [CHARTER.md](https://github.com/holoviz/holoviz/blob/main/doc/governance/org-docs/CHARTER.md).
+
+## HEP Format
+
+
+### Title
+All HEPs should begin with markdown title-level heading in the format of '# HEP {Identifier} - {Title}'. The identifier should remain a placeholder until a core developer has reviewed the pull request and assigned a number. The `Title` should match the entry in the table below.
+
+### Table
+Next, each HEP should include a table with the following information:
+
+<table>
+<tr><td> Identifier </td><td> [Number to be assigned by HoloViz core developer upon HEP acceptance] </td>
+<tr><td> Title </td><td> A short title of the proposal </td>
+<tr><td> Status </td><td> Draft | Proposed | Accepted | Rejected | Deferred | Implemented | Superseded </td></tr>
+<tr><td> Author(s) </td><td> Full Name [email is optional] </td></tr>
+<tr><td> Created </td><td> March 15, 2024</td></tr>
+<tr><td> Updated </td><td> March 15, 2024</td></tr>
+<tr><td> Discussion </td><td> link to the issue where the HEP is being discussed, NA before submission </td></tr>
+<tr><td> Implementation </td><td> link to the PR for the implementation, NA if not available </td></tr>
+</table>
+
+### Summary
+The table should be followed by a **Summary** section.
+
+### Specification
+If a HEP is perscriptive in nature (providing explicit guidelines or rules to follow), a **Specification** or **Policy** section is required to convey the technical details of the proposed enhancement.
+
+### Optional Sections
+Some sections that may be
+included if appropriate for the proposal include:
+
+- **Motivation/Background**: Why the proposed change or formalization is needed.
+- **Rationale**: Why particular decisions were made in the proposal.
+- **Backwards Compatibility**: Will the proposed change break existing packages or workflows.
+- **Alternatives**: Any alternatives considered during the design.
+- **Sample Implementation/Illustration**: Links to prototype or a sample implementation of the proposed change.
+- **FAQ**: Frequently asked questions (and answers to them).
+- **Resolution**: A short summary of the decision process conducted with the community.
+- **Reference**: Any references used in the design of the HEP.
+
```

Comment:
Maybe **Enforcement** (especially encouraging automation of enforcement)? 

I suspect a lot of HEPs are proposals that we agree on but may forget to enforce appropriately if no mechanism is in place to ensure compliance. Maybe the default policy is 'memorize and review all HEPs before release' but that seems to be a lot to ask of maintainers imho...

Reply:
good idea. I added a 'compliance' suggested section.